federatedscope/core/auxiliaries/data_builder.py

```diff
@@ -119,6 +119,27 @@ def _generate_data(client_num=5,
 
 
 def load_external_data(config=None):
+    r""" Based on the configuration file, this function imports external
+    datasets and applies train/valid/test splits and split by some specific `splitter`
+    into the standard FederatedScope input data format.
+
+    Args:
+        config: `CN` from `federatedscope/core/configs/config.py`
+
+    Returns:
+        data_local_dict: dict of split dataloader.
+                        Format:
+                            {
+                                'client_id': {
+                                    'train': DataLoader(),
+                                    'test': DataLoader(),
+                                    'val': DataLoader()
+                                }
+                            }
+        config: `CN` from `federatedscope/core/configs/config.py`
```

Comment:
It is confusing for users why we take a `config`and return a `config`. It would be better by naming it `modifed_cfg` with more informative explanation.

Other modifications look good to me